binding/rust/tests/rhino_tests.rs

```diff
@@ -95,216 +108,39 @@ mod tests {
-            "色変更",
-            HashMap::from([("色", "青")]),
-        ),
-        test_out_of_context_ja: (
-            "ja",
-            &"sumāto_shōmei",
-            false,
-            "色変更",
-            HashMap::new(),
-        ),
-        test_within_context_ko: (
-            "ko",
-            &"seumateu_jomyeong",
-            true,
-            "changeColor",
-            HashMap::from([("color", "파란색")]),
-        ),
-        test_out_of_context_ko: (
-            "ko",
-            &"seumateu_jomyeong",
-            false,
-            "changeColor",
-            HashMap::new(),
-        ),
-        test_within_context_pt: (
-            "pt",
-            &"luz_inteligente",
-            true,
-            "ligueLuz",
-            HashMap::from([("lugar", "cozinha")]),
-        ),
-        test_out_of_context_pt: (
-            "pt",
-            &"luz_inteligente",
-            false,
-            "ligueLuz",
-            HashMap::new(),
-        ),
-    }
-
     #[test]
     fn test_within_context() {
-        let access_key = env::var("PV_ACCESS_KEY")
-            .expect("Pass the AccessKey in using the PV_ACCESS_KEY env variable");
-        let context_path = format!(
-            "../../resources/contexts/{}/coffee_maker_{}.rhn",
-            pv_platform(),
-            pv_platform(),
-        );
-        let rhino = RhinoBuilder::new(access_key, context_path)
-            .init()
-            .expect("Unable to create Rhino");
-
-        let soundfile = BufReader::new(
-            File::open(concat!(
-                env!("CARGO_MANIFEST_DIR"),
-                "/../../resources/audio_samples/test_within_context.wav"
-            ))
-            .unwrap(),
-        );
-        let source = Decoder::new(soundfile).unwrap();
-
-        assert_eq!(rhino.sample_rate(), source.sample_rate());
-
-        let mut is_finalized = false;
-        for frame in &source.chunks(rhino.frame_length() as usize) {
-            let frame = frame.collect_vec();
-            if frame.len() == rhino.frame_length() as usize {
-                is_finalized = rhino.process(&frame).unwrap();
-                if is_finalized {
-                    break;
-                }
-            }
+        let test_json: Value = load_test_data();
+
+        for t in test_json["tests"]["within_context"].as_array().unwrap() {
+            let language = t["language"].as_str().unwrap();
+            let context = t["context_name"].as_str().unwrap();
+            let intent = t["inference"]["intent"].as_str().unwrap();
+            let slots_json = t["inference"]["slots"].as_object().unwrap();
+            let mut slots = HashMap::new();
+            slots_json.iter().for_each(|(key, value)| {
+                slots.insert(key.to_string(), String::from(value.as_str().unwrap()));
+            });
+
+            let test_audio = append_lang("test_within_context", language) + ".wav";
+            run_rhino_test(language, context, true, intent, slots, &test_audio);
```

Comment:
what does the failure look like if one language fails and the others don't? I seem to remember using the `Result` thing because it was much better at reporting individual errors.

Reply:
The built in `assert!` macros support adding custom messages, so I'll do that instead?